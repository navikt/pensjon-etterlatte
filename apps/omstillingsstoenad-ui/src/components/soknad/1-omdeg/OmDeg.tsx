import SoknadSteg from '../../../typer/SoknadSteg'
import { useTranslation } from 'react-i18next'
import InnloggetBruker from './InnloggetBruker'
import { SkjemaGruppe } from '../../felles/SkjemaGruppe'
import { FormProvider, useForm } from 'react-hook-form'
import { IValg } from '../../../typer/Spoersmaal'
import { useSoknadContext } from '../../../context/soknad/SoknadContext'
import { ISoeker } from '../../../typer/person'
import { ActionTypes } from '../../../context/soknad/soknad'
import { RHFInput, RHFKontonummerInput, RHFTelefonInput } from '../../felles/rhf/RHFInput'
import { RHFInlineRadio, RHFSpoersmaalRadio } from '../../felles/rhf/RHFRadio'
import Feilmeldinger from '../../felles/Feilmeldinger'
import { useBrukerContext } from '../../../context/bruker/BrukerContext'
import Navigasjon from '../../felles/Navigasjon'
import { Cell, Grid, Heading, RadioProps } from '@navikt/ds-react'
import { BankkontoType } from '../../../typer/utbetaling'
import UtenlandskBankInfo from './utenlandskBankInfo/UtenlandskBankInfo'
import HvorforSpoerVi from '../../felles/HvorforSpoerVi'
import { deepCopy } from '../../../utils/deepCopy'
import { RHFSelect } from '../../felles/rhf/RHFSelect'
import { useLand } from '../../../hooks/useLand'
import { SkjemaElement } from '../../felles/SkjemaElement'
import Bredde from '../../../typer/bredde'

const OmDeg: SoknadSteg = ({ neste }) => {
    const { t } = useTranslation()
    const { state, dispatch } = useSoknadContext()
    const brukerState = useBrukerContext().state
    const { land }: { land: any } = useLand()
    const lagre = (data: ISoeker) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG, payload: { ...deepCopy(data), erValidert: true } })
        neste!!()
    }

    const methods = useForm<ISoeker>({
        defaultValues: state.omDeg || {},
        shouldUnregister: true,
    })

    const {
        handleSubmit,
        watch,
        formState: { errors },
    } = methods

    const skalSjekkeFlyktningStatus = brukerState.foedselsaar!! < 1960

    const oppholderSegINorge = watch('oppholderSegINorge')
    const bankkontoType = watch('utbetalingsInformasjon.bankkontoType')

    return (
        <>
            <SkjemaElement>
                <Heading size={'medium'} className={'center'}>
                    {t('omDeg.tittel')}
                </Heading>
            </SkjemaElement>
            <InnloggetBruker />

            <FormProvider {...methods}>
                {/* TODO: Flytte dette til start eller eget steg? */}

                <form>
                    <SkjemaGruppe>
                        {!brukerState.adressebeskyttelse && !brukerState.adresse && (
                            <SkjemaGruppe>
                                <RHFInput name={'alternativAdresse'} label={t('omDeg.alternativAdresse')} />
                            </SkjemaGruppe>
                        )}

                        {!brukerState.telefonnummer && (
                            <SkjemaGruppe>
                                <Grid>
                                    <Cell xs={12} md={6} className={'kol'}>
                                        <RHFTelefonInput
                                            htmlSize={Bredde.S}
                                            name={'kontaktinfo.telefonnummer'}
                                            label={t('omDeg.kontaktinfo.telefonnummer')}
                                            valgfri={true}
                                        />
                                    </Cell>
                                </Grid>
                            </SkjemaGruppe>
                        )}
                    </SkjemaGruppe>

                    {/* 2.7 */}
                    {!brukerState.adressebeskyttelse && (
                        <SkjemaGruppe>
                            <SkjemaElement>
                                <RHFSpoersmaalRadio
                                    name={'oppholderSegINorge'}
                                    legend={t('omDeg.oppholderSegINorge')}
                                    description={
                                        <HvorforSpoerVi title="omDeg.oppholderSegINorge">
                                            {t('omDeg.oppholdHvorfor')}
                                        </HvorforSpoerVi>
                                    }
                                />
                            </SkjemaElement>

                            {oppholderSegINorge === IValg.JA && (
                                <SkjemaGruppe>
                                    <RHFKontonummerInput
                                        htmlSize={Bredde.S}
                                        name={'utbetalingsInformasjon.kontonummer'}
                                        label={t('omDeg.utbetalingsInformasjon.kontonummer')}
                                        placeholder={t('felles.elleveSiffer')}
                                        description={t('omDeg.utbetalingsInformasjon.informasjon')}
                                    />
                                </SkjemaGruppe>
                            )}

                            {oppholderSegINorge === IValg.NEI && (
                                <>
                                    <SkjemaGruppe>
                                        <RHFSelect
                                            className="kol-50"
                                            name={`oppholdsland`}
                                            label={t('omDeg.oppholdsland')}
                                            selectOptions={land}
                                        />
                                    </SkjemaGruppe>

                                    <SkjemaElement>
                                        <RHFInlineRadio
                                            name={'utbetalingsInformasjon.bankkontoType'}
                                            legend={t('omDeg.utbetalingsInformasjon.bankkontoType')}
                                        >
                                            {Object.values(BankkontoType).map((value) => {
                                                return { children: t(value), value } as RadioProps
                                            })}
                                        </RHFInlineRadio>
                                    </SkjemaElement>

                                    {bankkontoType === BankkontoType.norsk && (
                                        <RHFKontonummerInput
                                            htmlSize={Bredde.S}
                                            name={'utbetalingsInformasjon.kontonummer'}
                                            label={t('omDeg.utbetalingsInformasjon.kontonummer')}
                                            placeholder={t('felles.elleveSiffer')}
                                        />
                                    )}

                                    {bankkontoType === BankkontoType.utenlandsk && <UtenlandskBankInfo />}
                                </>
                            )}
                        </SkjemaGruppe>
                    )}

                    {skalSjekkeFlyktningStatus && (
                        <SkjemaGruppe>
                            <RHFSpoersmaalRadio
                                name={'flyktning'}
                                legend={t('omDeg.flyktning')}
                                description={
                                    <HvorforSpoerVi title="omDeg.flyktning">
                                        {t('omDeg.flyktningHvorfor')}
                                    </HvorforSpoerVi>
                                }
                            />
                        </SkjemaGruppe>
                    )}

                    <br />

                    <Feilmeldinger errors={errors} />

                    <Navigasjon neste={{ onClick: handleSubmit(lagre) }} />
                </form>
            </FormProvider>
        </>
    )
}

export default OmDeg
