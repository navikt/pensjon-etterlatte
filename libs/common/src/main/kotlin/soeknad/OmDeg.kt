package soeknad

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonValue
import no.nav.etterlatte.libs.common.soeknad.Valg

@JsonIgnoreProperties(ignoreUnknown = true)
data class OmDeg(
    val bostedsadresseBekreftet: Valg?,
    val alternativAdresse: String?,
    val kontaktinfo: Kontaktinfo?,
    val utbetalingsInformasjon: UtbetalingsInformasjon?,
    val flyktning: Valg?,
    val oppholderSegINorge: Valg?,
    val oppholdsland: String?,
    val medlemFolketrygdenUtland: Valg?,
    val nySivilstatus: NySivilstatus?
)

data class Kontaktinfo(
    val epost: String?,
    val telefonnummer: String?
)

data class UtbetalingsInformasjon(
    val kontonummer: String?,
    val bankkontoType: BankkontoType?,
    val utenlandskBankNavn: String?,
    val utenlandskBankAdresse: String?,
    val iban: String?,
    val swift: String?
)

enum class BankkontoType(@get:JsonValue val beskrivelse: String) {
    norsk("bankkontoType.norsk"),
    utenlandsk("bankkontoType.utenlandsk")
}
