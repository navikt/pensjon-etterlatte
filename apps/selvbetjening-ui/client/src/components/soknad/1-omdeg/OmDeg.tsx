import "./OmDeg.scss";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useTranslation } from "react-i18next";
import InnloggetBruker from "./InnloggetBruker";
import { RadioProps, SkjemaGruppe } from "nav-frontend-skjema";
import { FormProvider, useForm } from "react-hook-form";
import { IValg } from "../../../typer/Spoersmaal";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { ISoeker } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { RHFInput, RHFKontonummerInput, RHFTelefonInput } from "../../felles/RHFInput";
import { RHFInlineRadio, RHFSpoersmaalRadio } from "../../felles/RHFRadio";
import Feilmeldinger from "../../felles/Feilmeldinger";
import { useBrukerContext } from "../../../context/bruker/BrukerContext";
import Navigasjon from "../../felles/Navigasjon";
import { emailMatcher } from "../../../utils/matchers";
import { Cell, Grid, Heading } from "@navikt/ds-react";
import { BankkontoType } from "../../../typer/utbetaling";
import UtenlandskBankInfo from "./utenlandskBankInfo/UtenlandskBankInfo";
import HvorforSpoerVi from "../../felles/HvorforSpoerVi";
import SkjemaGruppering from "../../felles/SkjemaGruppering";
import { deepCopy } from "../../../utils/deepCopy";
import { RHFSelect } from "../../felles/RHFSelect";
import { useLand } from "../../../hooks/useLand";

const OmDeg: SoknadSteg = ({ neste }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();
    const brukerState = useBrukerContext().state;
    const { land }: { land: any } = useLand();
    const lagre = (data: ISoeker) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG, payload: { ...deepCopy(data), erValidert: true } });
        neste!!();
    };

    const methods = useForm<ISoeker>({
        defaultValues: state.omDeg || {},
        shouldUnregister: true,
    });

    const {
        handleSubmit,
        watch,
        formState: { errors },
    } = methods;

    const skalSjekkeFlyktningStatus = brukerState.foedselsaar!! < 1960;

    const borPaaRegistrertAdresse = watch("bostedsadresseBekreftet");
    const oppholderSegINorge = watch("oppholderSegINorge");
    const bankkontoType = watch("utbetalingsInformasjon.bankkontoType");

    return (
        <>
            {/* Steg 2 */}
            <Heading size={"medium"} className={"center"}>
                {t("omDeg.tittel")}
            </Heading>

            {/* Informasjon om den innloggede brukeren */}
            <InnloggetBruker/>

            {/* Skjema for utfylling av info om innlogget bruker / s√∏ker */}
            <FormProvider {...methods}>
                {/* TODO: Flytte dette til start eller eget steg? */}

                <form>
                    <SkjemaGruppering>
                        {!brukerState.adressebeskyttelse && (
                            <>
                                <RHFSpoersmaalRadio
                                    name={"bostedsadresseBekreftet"}
                                    legend={t("omDeg.bostedsadresseBekreftet")}
                                />

                                {borPaaRegistrertAdresse === IValg.NEI && (
                                    <SkjemaGruppe>
                                        <RHFInput
                                            name={"alternativAdresse"}
                                            label={t("omDeg.alternativAdresse")}
                                        />
                                    </SkjemaGruppe>
                                )}
                            </>
                        )}

                        <SkjemaGruppe>
                            <Grid>
                                <Cell xs={12} md={3} className={"kol"}>
                                    <RHFTelefonInput
                                        bredde={"S"}
                                        name={"kontaktinfo.telefonnummer"}
                                        label={t("omDeg.kontaktinfo.telefonnummer")}
                                        placeholder={t("omDeg.kontaktinfo.telefonnummerPlaceholder")}
                                        valgfri={true}
                                    />
                                </Cell>

                                <Cell xs={12} md={6} className={"kol"}>
                                    <RHFInput
                                        name={"kontaktinfo.epost"}
                                        label={t("omDeg.kontaktinfo.epost")}
                                        rules={{ pattern: emailMatcher }}
                                    />
                                </Cell>
                            </Grid>
                        </SkjemaGruppe>
                    </SkjemaGruppering>

                    {/* 2.7 */}
                    {!brukerState.adressebeskyttelse && (
                        <SkjemaGruppering>
                            <RHFSpoersmaalRadio
                                name={"oppholderSegINorge"}
                                legend={t("omDeg.oppholderSegINorge")}
                                description={
                                    <HvorforSpoerVi title="omDeg.oppholderSegINorge">
                                        {t("omDeg.oppholdHvorfor")}
                                    </HvorforSpoerVi>
                                }
                            />

                            {oppholderSegINorge === IValg.JA && (
                                <SkjemaGruppe>
                                    <RHFKontonummerInput
                                        bredde={"S"}
                                        name={"utbetalingsInformasjon.kontonummer"}
                                        label={t("omDeg.utbetalingsInformasjon.kontonummer")}
                                        placeholder={"11 siffer"}
                                        description={t("omDeg.utbetalingsInformasjon.informasjon")}
                                    />
                                </SkjemaGruppe>
                            )}

                            {oppholderSegINorge === IValg.NEI && (
                                <>
                                    <SkjemaGruppe>
                                        <RHFSelect
                                            className="kol-50"
                                            name={`oppholdsland`}
                                            label={t("omDeg.oppholdsland")}
                                            selectOptions={land}
                                        />
                                    </SkjemaGruppe>

                                    <RHFSpoersmaalRadio
                                        name={"medlemFolketrygdenUtland"}
                                        legend={t("omDeg.medlemFolketrygdenUtland")}
                                    />

                                    <RHFInlineRadio
                                        name={"utbetalingsInformasjon.bankkontoType"}
                                        legend={t("omDeg.utbetalingsInformasjon.bankkontoType")}
                                        radios={Object.values(BankkontoType).map((value) => {
                                            return { label: t(value), value } as RadioProps;
                                        })}
                                    />

                                    {bankkontoType === BankkontoType.norsk && (
                                        <RHFKontonummerInput
                                            bredde={"S"}
                                            name={"utbetalingsInformasjon.kontonummer"}
                                            label={t("omDeg.utbetalingsInformasjon.kontonummer")}
                                            placeholder={"11 siffer"}
                                        />
                                    )}

                                    {bankkontoType === BankkontoType.utenlandsk && <UtenlandskBankInfo/>}
                                </>
                            )}
                        </SkjemaGruppering>
                    )}

                    {skalSjekkeFlyktningStatus && (
                        <SkjemaGruppe>
                            <RHFSpoersmaalRadio
                                name={"flyktning"}
                                legend={t("omDeg.flyktning")}
                                description={
                                    <HvorforSpoerVi title="omDeg.flyktning">
                                        {t("omDeg.flyktningHvorfor")}
                                    </HvorforSpoerVi>
                                }
                            />
                        </SkjemaGruppe>
                    )}

                    <br/>

                    <Feilmeldinger errors={errors}/>

                    <Navigasjon neste={{ onClick: handleSubmit(lagre) }}/>
                </form>
            </FormProvider>
        </>
    );
};

export default OmDeg;
