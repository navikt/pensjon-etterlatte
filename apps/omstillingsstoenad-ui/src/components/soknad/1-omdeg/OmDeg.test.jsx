import { render, fireEvent } from "@testing-library/react";
import OmDeg from "./OmDeg";
import { describe, expect, it, vi } from 'vitest'
import {BrowserRouter} from "react-router-dom";

vi.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    ...vi.importActual("react-i18next"),
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

// Her må det mockes litt forskjellige guid-pakker...
vi.mock("uuid", () => ({
    v4: () => "456"
}))

describe("Om deg og avdød", () => {
    it("Snapshot", () => {
        const { container, getByText } = render(<BrowserRouter><OmDeg /></BrowserRouter>);
        
        const bankkontoTypeUtenlandsk = container.querySelectorAll("[value='bankkontoType.utenlandsk']")[0];
        fireEvent.click(bankkontoTypeUtenlandsk);

        expect(getByText("omDeg.utbetalingsInformasjon.iban")).toBeDefined();
        expect(container).toMatchSnapshot();
    });
});
