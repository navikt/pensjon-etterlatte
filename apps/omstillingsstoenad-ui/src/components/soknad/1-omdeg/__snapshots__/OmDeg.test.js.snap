// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om deg og avdød Snapshot 1`] = `
<div>
  <div
    class="sc-eTNRI cTBPfi"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDeg.tittel
    </h1>
  </div>
  <form>
    <fieldset
      class="sc-beySPh hBKmDL"
    >
      <fieldset
        class="sc-beySPh hBKmDL"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="alternativAdresse"
            >
              omDeg.alternativAdresse
            </label>
            <input
              class="navds-text-field__input navds-body-short navds-body-short--medium"
              id="alternativAdresse"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-r0"
            />
          </div>
        </div>
      </fieldset>
      <fieldset
        class="sc-beySPh hBKmDL"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="kontaktinfo.telefonnummer"
          >
            omDeg.kontaktinfo.telefonnummer
          </label>
          <input
            class="navds-text-field__input navds-body-short navds-body-short--medium"
            id="kontaktinfo.telefonnummer"
            size="25"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-r1"
          />
        </div>
      </fieldset>
    </fieldset>
    <fieldset
      class="sc-beySPh hBKmDL"
    >
      <div
        class="sc-eTNRI cTBPfi"
      >
        <div
          id="utbetalingsInformasjon.bankkontoType"
        >
          <fieldset
            class="navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
          >
            <legend
              class="navds-fieldset__legend navds-label"
            >
              omDeg.utbetalingsInformasjon.bankkontoType
            </legend>
            <div
              class="navds-radio-buttons"
            >
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  class="navds-radio__input"
                  id="radio-r4"
                  name="utbetalingsInformasjon.bankkontoType"
                  type="radio"
                  value="bankkontoType.norsk"
                />
                <label
                  class="navds-radio__label"
                  for="radio-r4"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short navds-body-short--medium"
                    >
                      bankkontoType.norsk
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  class="navds-radio__input"
                  id="radio-r5"
                  name="utbetalingsInformasjon.bankkontoType"
                  type="radio"
                  value="bankkontoType.utenlandsk"
                />
                <label
                  class="navds-radio__label"
                  for="radio-r5"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short navds-body-short--medium"
                    >
                      bankkontoType.utenlandsk
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-fieldset__error"
              id="fieldset-error-r3"
            />
          </fieldset>
        </div>
      </div>
      <div
        class="sc-eTNRI cTBPfi"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDeg.utbetalingsInformasjon.tittel
        </h1>
      </div>
      <div
        class="sc-eTNRI cTBPfi"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="utbetalingsInformasjon.utenlandskBankNavn"
            >
              omDeg.utbetalingsInformasjon.utenlandskBankNavn
            </label>
            <input
              class="navds-text-field__input navds-body-short navds-body-short--medium"
              id="utbetalingsInformasjon.utenlandskBankNavn"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-r9"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-eTNRI cTBPfi"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="utbetalingsInformasjon.utenlandskBankAdresse"
            >
              omDeg.utbetalingsInformasjon.utenlandskBankAdresse
            </label>
            <input
              class="navds-text-field__input navds-body-short navds-body-short--medium"
              id="utbetalingsInformasjon.utenlandskBankAdresse"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-ra"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-eTNRI cTBPfi"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="utbetalingsInformasjon.iban"
          >
            <div
              class="sc-brSamD dpbKKW"
            >
              omDeg.utbetalingsInformasjon.iban
               
              <div
                class="navds-help-text"
              >
                <button
                  aria-expanded="false"
                  class="navds-help-text__button"
                  type="button"
                >
                  <svg
                    aria-labelledby="title-rc"
                    class="navds-help-text__icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rc"
                    >
                      hjelp
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="transparent"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <svg
                    aria-labelledby="title-rd"
                    class="navds-help-text__icon navds-help-text__icon--filled"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rd"
                    >
                      hjelp
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="currentColor"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="var(--a-surface-default)"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <div
                  aria-hidden="true"
                  class="navds-popover navds-help-text__popover navds-popover--hidden"
                  data-placement="top"
                  style="position: absolute; top: 0px; left: 0px;"
                >
                  <div
                    class="navds-popover__content navds-body-short"
                  >
                    omDeg.utbetalingsInformasjon.ibanHjelpetekst
                  </div>
                  <div
                    class="navds-popover__arrow"
                    style="bottom: -0.5rem;"
                  />
                </div>
              </div>
            </div>
          </label>
          <input
            class="navds-text-field__input navds-body-short navds-body-short--medium"
            id="utbetalingsInformasjon.iban"
            size="35"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-rb"
          />
        </div>
      </div>
      <div
        class="sc-eTNRI cTBPfi"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="utbetalingsInformasjon.swift"
          >
            <div
              class="sc-brSamD dpbKKW"
            >
              omDeg.utbetalingsInformasjon.swift
               
              <div
                class="navds-help-text"
              >
                <button
                  aria-expanded="false"
                  class="navds-help-text__button"
                  type="button"
                >
                  <svg
                    aria-labelledby="title-rg"
                    class="navds-help-text__icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rg"
                    >
                      hjelp
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="transparent"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <svg
                    aria-labelledby="title-rh"
                    class="navds-help-text__icon navds-help-text__icon--filled"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rh"
                    >
                      hjelp
                    </title>
                    <circle
                      cx="12"
                      cy="12"
                      fill="currentColor"
                      r="11"
                      stroke="currentColor"
                      stroke-width="1.5"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M9.75 9C9.75 7.75736 10.7574 6.75 12 6.75H12.1716C13.3195 6.75 14.25 7.68054 14.25 8.82843C14.25 9.37966 14.031 9.90832 13.6412 10.2981L12.6412 11.2981C11.7504 12.1889 11.25 13.3971 11.25 14.6569C11.25 15.0711 11.5858 15.4069 12 15.4069C12.4142 15.4069 12.75 15.0711 12.75 14.6569C12.75 13.7949 13.0924 12.9682 13.7019 12.3588L14.7019 11.3588C15.373 10.6877 15.75 9.77748 15.75 8.82843C15.75 6.85212 14.1479 5.25 12.1716 5.25H12C9.92893 5.25 8.25 6.92893 8.25 9V9.5C8.25 9.91421 8.58579 10.25 9 10.25C9.41421 10.25 9.75 9.91421 9.75 9.5V9ZM12 16.5C11.4477 16.5 11 16.9477 11 17.5C11 18.0523 11.4477 18.5 12 18.5C12.5523 18.5 13 18.0523 13 17.5C13 16.9477 12.5523 16.5 12 16.5Z"
                      fill="var(--a-surface-default)"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <div
                  aria-hidden="true"
                  class="navds-popover navds-help-text__popover navds-popover--hidden"
                  data-placement="top"
                  style="position: absolute; top: 0px; left: 0px;"
                >
                  <div
                    class="navds-popover__content navds-body-short"
                  >
                    omDeg.utbetalingsInformasjon.swiftHjelpetekst
                  </div>
                  <div
                    class="navds-popover__arrow"
                    style="bottom: -0.5rem;"
                  />
                </div>
              </div>
            </div>
          </label>
          <input
            class="navds-text-field__input navds-body-short navds-body-short--medium"
            id="utbetalingsInformasjon.swift"
            size="25"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-rf"
          />
        </div>
      </div>
    </fieldset>
    <br />
    <fieldset
      class="sc-beySPh hBKmDL"
    >
      <div
        class="sc-guDLey efPyOT undefined"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            class="navds-label"
          >
            knapp.neste
          </span>
        </button>
      </div>
    </fieldset>
    <div
      class="sc-guDLey sc-dmyCSP efPyOT gPrijY"
    >
      <button
        class="navds-button navds-button--tertiary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          class="navds-label"
        >
          knapp.avbryt
        </span>
      </button>
    </div>
    <dialog
      class="navds-modal navds-modal--polyfilled navds-modal--autowidth"
      role="dialog"
    >
      <div
        class="navds-modal__header"
      >
        <button
          class="navds-modal__button navds-button navds-button--tertiary-neutral navds-button--small navds-button--icon-only"
          type="button"
        >
          <span
            class="navds-button__icon"
          >
            <svg
              aria-labelledby="title-r8"
              fill="none"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title-r8"
              >
                Lukk modalvindu
              </title>
              <path
                d="M6.53 5.47a.75.75 0 0 0-1.06 1.06L10.94 12l-5.47 5.47a.75.75 0 1 0 1.06 1.06L12 13.06l5.47 5.47a.75.75 0 1 0 1.06-1.06L13.06 12l5.47-5.47a.75.75 0 0 0-1.06-1.06L12 10.94 6.53 5.47Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <h1
          class="navds-heading navds-heading--medium"
        >
          avbrytModal.spoersmaal
        </h1>
      </div>
      <div
        class="navds-modal__body"
      >
        <p
          class="mute avbryt-text navds-body-short navds-body-short--medium"
        >
          avbrytModal.informasjon
        </p>
      </div>
      <div
        class="navds-modal__footer"
      >
        <div
          class="sc-jsEeTM buOcVS"
        >
          <button
            class="navds-button navds-button--secondary navds-button--medium"
            id="avbryt-nei-btn"
            style="margin: 10px;"
            type="button"
          >
            <span
              class="navds-label"
            >
              avbrytModal.svarNei
            </span>
          </button>
          <button
            class="navds-button navds-button--primary navds-button--medium"
            id="avbryt-ja-btn"
            style="margin: 10px;"
            type="button"
          >
            <span
              class="navds-label"
            >
              avbrytModal.svarJa
            </span>
          </button>
        </div>
        <div
          class="sc-jsEeTM buOcVS"
        >
          <button
            class="navds-button navds-button--tertiary navds-button--medium"
            id="slett-soeknad"
            style="color: rgb(198, 93, 78);"
            type="button"
          >
            <span
              class="navds-label"
            >
              avbrytModal.svarSlett
            </span>
          </button>
        </div>
      </div>
    </dialog>
  </form>
</div>
`;
